src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,171 @@
+package chatbot;
+
+import java.util.List;
+
+/**
+ * Handles the interpretation of user input.
+ */
+public class Parser {
+    /** UI to handle the printing of output. */
+    private UI ui;
+    /** taskHandler to store the list of Tasks and do operations. */
+    private TaskList taskHandler;
+    /** Storage to load and write files. */
+    private Storage storage;
+
+    /**
+     * Constructor for Parser. Initialises the UI, taskHandler, storage.
+     */
+    public Parser(UI ui, TaskList taskHandler, Storage storage) {
+        this.ui = ui;
+        this.taskHandler = taskHandler;
+        this.storage = storage;
+    }
+
+    // Purely for testing purposes.
+    public Parser() {}
+
+    /** 
+     * Interprets the user string input and
+     * returns a value based on the input.
+     * 
+     * @param input The user input to be interpreted.
+     * @return The value to determine if the program ends. 
+     */
+    public String runInput(String input) {
```

Comment:
Code here for this method is slightly long. Perhaps could consider implementing a command class?